src/transaction/lock.rs

```diff
@@ -21,67 +22,76 @@ pub async fn resolve_locks(
     pd_client: Arc<impl PdClient>,
 ) -> Result<bool> {
     let ts = pd_client.clone().get_timestamp().await?;
+    type LockVersion = u64;
     let mut has_live_locks = false;
-    let expired_locks = locks.into_iter().filter(|lock| {
+
+    struct ResolveInfo {
+        txn_size: u64,
+        region: Region,
+        keys: Vec<Key>,
```

Comment:
I think we can save primary key only once in `ResolveInfo`. And the `keys` are used to save the lock keys if the transaction is small enough.